src/main/java/com/exactpro/th2/common/schema/factory/AbstractCommonFactory.java

```diff
@@ -105,6 +107,8 @@
     private final AtomicReference<MessageRouterConfiguration> messageRouterConfiguration = new AtomicReference<>();
     private final AtomicReference<GrpcRouterConfiguration> grpcRouterConfiguration = new AtomicReference<>();
 
+    private final ScheduledExecutorService executor = new ScheduledThreadPoolExecutor(3);
```

Comment:
Why do we use 3 threads? Maybe it will be configurable with the default value 1.